packages/flutter/lib/src/widgets/safe_area.dart

```diff
@@ -98,7 +98,7 @@ class SafeArea extends StatelessWidget {
     final MediaQueryData data = MediaQuery.of(context);
     EdgeInsets padding = data.padding;
     // Bottom padding has been consumed - i.e. by the keyboard
-    if (data.padding.bottom == 0.0 && data.viewInsets.bottom != 0.0 && maintainBottomViewPadding)
+    if (padding.bottom == 0.0 && data.viewInsets.bottom != 0.0 && maintainBottomViewPadding)
```

Comment:
I think this is meant to be more readable the way it is. We make a copy of the padding, and then based on the data we may update that copy. I am not sure that we need to make this change.

Reply:
In line 102, `padding` is used instead of `data.padding`  so I thought it would be better to maintain that throughout
Anyways, you can close the PR if you feel, theirs no need for it.